src/librustc_mir/borrow_check/borrow_set.rs

```diff
@@ -150,7 +150,7 @@ struct GatherBorrows<'a, 'gcx: 'tcx, 'tcx: 'a> {
     pending_activations: FxHashMap<mir::Local, BorrowIndex>,
 }
 
-impl<'a, 'gcx, 'tcx> Visitor<'tcx> for GatherBorrows<'a, 'gcx, 'tcx> {
+impl Visitor<'tcx> for GatherBorrows<'a, 'gcx, 'tcx> {
```

Comment:
We don't necessarily want to make such changes in *this* PR, but I think that the conventions I would like is that you must use multi-letter names on impls -- single-letter names are only appropriate for functions. In other words, if you see a single-letter name, you know it is local in scope and don't have to look further. If you see a multi-letter name, it may match up with something on the impl. (The reason I don't say *must* match up is that I want us to be able to write `'tcx` anywhere.)

This would mean that instead of `'a` we should use something else; I typically use `'cx` but I now think we should use `'me` or `'this`